src/auto-trace.js

```diff
@@ -111,6 +149,10 @@ function executeAsyncMiddleware(asyncErr) {
  * @return {Error} the new error, after it has been passed through the middlewares
  */
 function executeSyncMiddleware(middlewares, ogErr) {
+	//Don't run middlewares on errors that have already been handled by auto-trace
+	if (ogErr && ogErr.autoTraceIgnore){
+		return ogErr
```

Comment:
executeAsyncMiddleware is called before an error happens. At this point we have only created an asyncStacktrace in case it is needed. We may never use this stacktrace (if no error occurs).  executeAsyncMiddleware is here to provide the middleware with the asyncStacktrace in case it is needed. 

executeSyncMiddleware is called once an actual error occurs. In this case we can examine the error to determine if it has already be handled by auto-trace.